### PR TITLE
DOC: Add descriptions of filter representations to scipy.signal tutorial

### PR DESCRIPTION
**Pull request by endolith**
for https://github.com/scipy/scipy/issues/3754

and some typos I found in the process

Not finished yet; please give advice.

### CODE REVIEW

**Comment by endolith** [comment]
Well it's not finished.  Anything else I should add?  Anything wrong?  I want to make an image to illustrate the frequency transformation.

http://dsp.stackexchange.com/q/13402/29

Should the discrete SS be written as 

```
\dot{\mathbf{x}} = \mathbf{A}\mathbf{x} + \mathbf{B}\mathbf{u}\\
\mathbf{y} = \mathbf{C}\mathbf{x} + \mathbf{D}\mathbf{u}
```
2015-01-05 19_19_39-texrendr com _ browser-based real-time tex_latex visualization write see sh

or

```
\mathbf{x}(k+1) = A \mathbf{x}(k) + B \mathbf{u}(k)\\
\mathbf{y}(k) = C \mathbf{x}(k) + D \mathbf{u}(k)
```
2015-01-05 19_20_08-texrendr com _ browser-based real-time tex_latex visualization write see sh

or x[n-1] or something like that?

**Comment by endolith** [comment]
Should clarify there are two tf forms, for positive or negative powers. Input to residue vs residuez for instance

**Comment by endolith** [comment]
rebased to avoid conflicts, but still has unfinished bits

**Comment by Eric89GXL** [comment]
>  want to make an image to illustrate the frequency transformation but I don't know how to include images.

I know how to do it in ReStructuredText, maybe it's the same. Can you look in the docs to see if there is an existing example to follow?

The only static image in the repo I see is https://github.com/scipy/scipy/blob/master/doc/source/_static/scipyshiny_small.png

> or x[n-1] or something like that?

I'd say go with whatever you see in your favorite textbook. I think I'm used to seeing `x[n] = ... x[n - 1]` for the discrete version, but I don't have a strong feeling.

> Anything else I should add?

Smaller PRs are fine, so we don't need to be complete here.

> Anything wrong?

Once you have it in a mergeable state, I'll try to look closer. For now I assume it's correct :)

**Comment by endolith** [comment]
Rebased and removed unfinished things

**Comment by Eric89GXL** [comment]
LGTM. @e-q want to have a look and merge if you're also happy?

File: doc/source/tutorial/signal.rst [hunk 1]
```diff
@@ -594,6 +594,168 @@ stop-band attenuation of :math:`\approx 60` dB.
...
+    \mathbf{x}[k+1] = A \mathbf{x}[k] + B \mathbf{u}[k]\\
+    \mathbf{y}[k] = C \mathbf{x}[k] + D \mathbf{u}[k]
+
+or a continuous/analog system of the form:
+
+.. math::
+    \dot{\mathbf{x}}(t) = A \mathbf{x}(t) + B \mathbf{u}(t)\\
+    \mathbf{y}(t) = C \mathbf{x}(t) + D \mathbf{u}(t)
+
+with `p` inputs, `q` outputs and `N` state variables, where:
```
Comment (e-q) [inline doc/source/tutorial/signal.rst]: `p` and `q` should be capitalized to match the text below

File: doc/source/tutorial/signal.rst [hunk 2]
```diff
@@ -594,6 +594,168 @@ stop-band attenuation of :math:`\approx 60` dB.
...
+    [b_0, b_1, b_2, a_0, a_1, a_2]
+
+The coefficients are typically normalized such that :math:`a_0` is always 1.
+The section order is usually not important with floating-point computation;
+the filter output will be the same regardless.
+
+Filter transformations
+""""""""""""""""""""""
+
+are transformed using the following substitutions:
```
Comment (e-q) [inline doc/source/tutorial/signal.rst]: Can you add a little more explanation here?

**Comment by e-q** [comment]
Other than the two little points I raised in the line comments, this looks nice.

File: doc/source/tutorial/signal.rst (hunk 2 above)
Comment (endolith) [inline doc/source/tutorial/signal.rst]: oops

**Comment by endolith** [comment]
updated

**Comment by Eric89GXL** [comment]
Thanks @endolith

**Pull request merged**